### ISSUE TITLE
Crash when starting the first scale "get values"

### ISSUE THREAD

**Issue by spfmoby** [lswiderski/mi-scale-exporter#6]
Device: Xiaomi Note 4X
Android version: 7.0 (latest for this smartphone) / MiUI 9.0
Scale properly connected to the Zepp life app.

How to produce the issue: start to fill the age, height, scale address... then click on the "get values from mi scale" button.

What happens: the app immediatly crashes with the following error:
```
android.runtime.JavaProxyThrowable: System.FormatException: Input string was not in a correct format.
  at System.Number.ThrowOverflowOrFormatException (System.Boolean overflow, System.String overflowResourceKey) [0x0001a] in <862eac71fdb24d8a944879927b29f0c0>:0 
  at System.Number.ParseDouble (System.ReadOnlySpan`1[T] value, System.Globalization.NumberStyles styles, System.Globalization.NumberFormatInfo info) [0x0007d] in <862eac71fdb24d8a944879927b29f0c0>:0 
  at System.Double.Parse (System.String s) [0x0001a] in <862eac71fdb24d8a944879927b29f0c0>:0 
  at MiScaleExporter.ViewModels.ScaleViewModel.OnScan () [0x0004f] in <ed6ca7ffd07c4c57acc3b8941b348ed6>:0 
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) [0x00000] in <862eac71fdb24d8a944879927b29f0c0>:0 
  at Android.App.SyncContext+<>c__DisplayClass2_0.<Post>b__0 () [0x00000] in <eedaa2491856441381e221d6410352cc>:0 
  at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <eedaa2491856441381e221d6410352cc>:0 
  at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00008] in <eedaa2491856441381e221d6410352cc>:0 
  at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.118(intptr,intptr)
	at mono.java.lang.RunnableImplementor.n_run(Native Method)
	at mono.java.lang.RunnableImplementor.run(RunnableImplementor.java:30)
	at android.os.Handler.handleCallback(Handler.java)
	at android.os.Handler.dispatchMessage(Handler.java)
	at android.os.Looper.loop(Looper.java)
	at android.app.ActivityThread.main(ActivityThread.java)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java)
```
The error happens every time
All the required fields are populated with correct values (integer for age and height...)

**Comment by lswiderski** [comment]
Hi, the problem with debugging older versions of android is that, the emulator doesn't work with bluetooth connections.
The error is throw on checking the system versions I modified, the code a little and maybe that can help in next release.

**Comment by lswiderski** [comment]
Please check it out https://github.com/lswiderski/mi-scale-exporter/releases/tag/v1.5

**Comment by spfmoby** [comment]
On the Xiaomi Mi Note 4, the behabiour is now a little bit different.
When I start "get values", the scan does not immediatly crashes. Is scans for a very long time (same as the other bug with my Samsung) then crashes with the following error:

```
android.runtime.JavaProxyThrowable: System.InvalidOperationException
  at System.Threading.Tasks.TaskCompletionSource`1[TResult].SetResult (TResult result) [0x00009] in <862eac71fdb24d8a944879927b29f0c0>:0 
  at MiScaleExporter.Services.ScaleService.TimeOuted (System.Object s, System.EventArgs e) [0x0000b] in <59d460916b3c4bd0af0b82d92f0c43f8>:0 
  at (wrapper delegate-invoke) <Module>.invoke_void_object_EventArgs(object,System.EventArgs)
  at Plugin.BLE.Abstractions.AdapterBase.StartScanningForDevicesAsync (System.Guid[] serviceUuids, System.Func`2[T,TResult] deviceFilter, System.Boolean allowDuplicatesKey, System.Threading.CancellationToken cancellationToken) [0x001c1] in <0aa4eab098134c5995307ece1b757bd7>:0 
  at MiScaleExporter.Services.ScaleService.GetBodyCompositonAsync (MiScaleExporter.Models.Scale scale, MiScaleExporter.Models.User user) [0x000c2] in <59d460916b3c4bd0af0b82d92f0c43f8>:0 
  at MiScaleExporter.ViewModels.ScaleViewModel.OnScan () [0x002e3] in <59d460916b3c4bd0af0b82d92f0c43f8>:0 
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) [0x00000] in <862eac71fdb24d8a944879927b29f0c0>:0 
  at Android.App.SyncContext+<>c__DisplayClass2_0.<Post>b__0 () [0x00000] in <eedaa2491856441381e221d6410352cc>:0 
  at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <eedaa2491856441381e221d6410352cc>:0 
  at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00008] in <eedaa2491856441381e221d6410352cc>:0 
  at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.118(intptr,intptr)
	at mono.java.lang.RunnableImplementor.n_run(Native Method)
	at mono.java.lang.RunnableImplementor.run(RunnableImplementor.java:30)
	at android.os.Handler.handleCallback(Handler.java)
	at android.os.Handler.dispatchMessage(Handler.java)
	at android.os.Looper.loop(Looper.java)
	at android.app.ActivityThread.main(ActivityThread.java)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java)
```

**Comment by lswiderski** [comment]
Ok so this issue is resolved. I found in basement old Motorola phone with Android 7.1.1 (this number could not be parsed to double). I could received data from xiaomi scale even without Zepp Life app installed. Just entered proper Bluetooth address and I gave location permission to app. The log which you provided says there was a timeout, so same as in issue #5 . 

I'm closing this one as resolved

About timeout lets discuss in #5
